### PR TITLE
Track and collect the comments without the callback.

### LINKED ISSUE #1381: Improve comment collection in the scanner
There is no need to transfer the collected comments via the side-channel, i.e. `addComment` callback.

### PR DESCRIPTION
**Pull request by ariya**
Fixes #1381

### CODE REVIEW

File: test/utils/evaluate-testcase.js [hunk 1]
```diff
@@ -203,6 +203,16 @@
         expected = JSON.stringify(tokens, null, 4);
 
         try {
+            // Some variations of the options (tolerant mode, to avoid premature error)
+            esprima.tokenize(code, { tolerant: true, comment: false, loc: false, range: false });
+            esprima.tokenize(code, { tolerant: true, comment: false, loc: false, range: true });
+            esprima.tokenize(code, { tolerant: true, comment: false, loc: true,  range: false });
+            esprima.tokenize(code, { tolerant: true, comment: false, loc: false, range: true });
```
Comment (mikesherov) [inline test/utils/evaluate-testcase.js]: `loc: true` here or it's redundant of two lines above.

File: test/utils/evaluate-testcase.js [hunk 2]
```diff
@@ -203,6 +203,16 @@
...
+            esprima.tokenize(code, { tolerant: true, comment: true,  loc: false, range: false });
+            esprima.tokenize(code, { tolerant: true, comment: true,  loc: false, range: true });
+            esprima.tokenize(code, { tolerant: true, comment: true,  loc: true,  range: false });
+            esprima.tokenize(code, { tolerant: true, comment: true,  loc: false, range: true });
```
Comment (mikesherov) [inline test/utils/evaluate-testcase.js]: `loc: true` here or it's redundant of two lines above.

**Comment by mikesherov** [comment]
Should we pull JSCS out until I or @hzoo can figure out what's going on there?

**Comment by mikesherov** [comment]
Aside from the test nits, :+1: really great work here.

**Comment by ariya** [comment]
> Should we pull JSCS out until I or @hzoo can figure out what's going on there?

Good idea, I reopened #1247 to track that.

File: test/utils/evaluate-testcase.js (hunk 1 above)
Comment (ariya) [inline test/utils/evaluate-testcase.js]: Good catch!

File: test/utils/evaluate-testcase.js (hunk 2 above)
Comment (ariya) [inline test/utils/evaluate-testcase.js]: Now that I spot it, I realize it's not needed (the real tokenizer test uses that setting).

**Pull request closed without merging**